File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque


def dfs(v):
    ans_dfs.append(v)
    visited[v] = 1

    for i in graph[v]:
        if visited[i] == 0:
            dfs(i)


def bfs(v):
    queue = deque()

    queue.append(v)
    ans_bfs.append(v)
    visited[v] = 1

    while queue:
        b = queue.popleft()
        for n in graph[b]:
            if visited[n] == 0:
                queue.append(n)
                ans_bfs.append(n)
                visited[n] = 1


N, M, V = map(int, input().split())
graph = [[] for _ in range(N+1)]

for i in range(M):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)

for i in range(1, N+1):
    graph[i].sort()

visited = [0] * (N+1)
ans_dfs = []

dfs(V)

visited = [0] * (N+1)
ans_bfs = []

bfs(V)

print(*ans_dfs)
print(*ans_bfs)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;

public class Main {
	static ArrayList<Integer>[] lists;
	static boolean[] visited;
	static StringBuilder sb = new StringBuilder();

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		
		int N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());
		int V = Integer.parseInt(st.nextToken());
		lists = new ArrayList[N + 1];
		for(int i = 0; i <= N; i++)
			lists[i] = new ArrayList<>();
		for(int i = 0; i < M; i++) {
			st = new StringTokenizer(br.readLine());
			int s = Integer.parseInt(st.nextToken());
			int e = Integer.parseInt(st.nextToken());
			lists[s].add(e);
			lists[e].add(s);
		}
		for(ArrayList list : lists)
			Collections.sort(list);
		
		visited = new boolean[N + 1];
		dfs(V);
		sb.append("\n");
		visited = new boolean[N + 1];
		bfs(V);
		System.out.println(sb.toString());
	}


	private static void dfs(int v) {
		visited[v] = true;
		sb.append(v + " ");
		for(int n : lists[v]) {
			if(!visited[n])
				dfs(n);
		}
	}
	
	private static void bfs(int v) {
		ArrayDeque<Integer> deque = new ArrayDeque<>();
		visited[v] = true;
		deque.add(v);
		sb.append(v + " ");
		
		while(!deque.isEmpty()) {
			int num = deque.poll();
			for(int n : lists[num]) {
				if(!visited[n]) {
					visited[n] = true;
					deque.add(n);
					sb.append(n + " ");
				}
			}
		}
	}
}